data/tutorials/gs_01_a_first_hour_with_ocaml.md

```diff
@@ -82,15 +100,43 @@ val x : int = 50
 - : int = 2500
 ```
 
-Note that this isn't a variable as in languages like C and Python. Its value
-cannot be changed. We can write it all in one go using `let` ... `in` ...:
+**Please note** that this variable is different from the variables of imperative
+languages (e.g., C or Python): *its value can never be changed*. In that sense,
+after being assigned a value, the variable behaves like a constant. In functional
+languages like OCaml, all variables are *immutable*. The consequence is that
+most functions you will write do not have side-effects (i.e. calling a function does not change
+the state of the overall system). However, there's a caveat here: side-effects
+can occur when performing IO, or when working with mutable state on the heap
+(the variable referencing the mutable heap state is immutable,
+but the state pointed to is not).
+
+A variable's name must begin with a lowercase letter or an underscore, and it cannot contain dashes (i.e., `let x-plus-y`).
+
+When we type `let x = 50;;` and Enter, OCaml confirms the variable's value (`val`), showing that "x" now means "50." So when we write the expression `x * x;;`, it will result in the same answer as `50 * 50;;`. 
```

Comment:
```suggestion
**Please note** that this variable is different from the variables of imperative
languages (e.g., C or Python): *its value can never be changed*. In that sense,
after being assigned a value, the variable behaves like a constant. In functional
languages like OCaml, all variables are *immutable*. The consequence is that
most functions you will write do not have side-effects (i.e., calling a function does not change
the state of the overall system). However, there's a caveat here: side-effects
can occur when performing I/O or when working with mutable state on the heap
(the variable referencing the mutable heap state is immutable,
but the state pointed to is not).

A variable's name must begin with a lowercase letter or an underscore, and it cannot contain dashes (i.e., `let x-plus-y`).

When we type `let x = 50;;` and Enter, OCaml confirms the variable's value (`val`), showing that "x" now means "50." So when we write the expression `x * x;;`, it will result in the same answer as `50 * 50;;`. 
```